core/domain/exp_services.py

```diff
@@ -1610,18 +1611,22 @@ def regenerate_exploration_and_contributors_summaries(exploration_id):
 
 
 def compute_summary_of_exploration(
-    exploration, exp_rights, exp_summary_model,
-    skip_exploration_model_last_updated=False):
+    exploration,
+    exp_rights,
+    exp_summary_model,
+    skip_exploration_model_last_updated=False
+):
     """Create an ExplorationSummary domain object for a given Exploration
     domain object and return it.
 
     Args:
         exploration: Exploration. The exploration whose summary is to be
             computed.
-        exp_rights: ExplorationRightsModel. The exploration rights model, used
+        exp_rights: ActivityRights. The exploration rights model, used
             to compute summary.
-        exp_summary_model: ExplorationSummaryModel. The exploration summary
-            model, whose summary is computed.
+        exp_summary_model: ExplorationSummaryMode|None. The exploration summary
+            model, whose summary is computed, can be None, it that case we
```

Comment:
The exploration summary model whose summary needs to be recomputed. If None, we generate a new exploration summary model instead.

Reply:
Done.